### PR TITLE
[Odysseus] Overhauling framework in line with backend services

### PR DESCRIPTION
**Pull request by AllTerrainDeveloper**
Related to  https://github.com/Automattic/ai-services/issues/59

## Proposed Changes

Overhauling the Odysseus Framework as per design specifications

## Testing Instructions

Those are specified in D115311 :)

### CODE REVIEW

File: client/odysseus/context/index.tsx [hunk 1]
```diff
@@ -9,25 +9,52 @@ export type Nudge = {
 	initialMessage: string;
 	context?: Record< string, unknown >;
 };
+
+export type MessageRole = 'user' | 'bot';
```
Comment (kthai15) [inline client/odysseus/context/index.tsx:13]: Updated this to match the enum declaration on the server
https://github.com/Automattic/ai-services/blob/trunk/odysseus/app/schemas.py#L15-L18

File: client/odysseus/context/index.tsx [hunk 2]
```diff
@@ -9,25 +9,52 @@ export type Nudge = {
...
+
+export type MessageType = 'message' | 'action' | 'meta' | 'error';
+
+export type Message = {
+	content: string;
+	role: MessageRole;
+	type: MessageType;
+	chatId?: string | null;
+};
```
Comment (kthai15) [inline client/odysseus/context/index.tsx:22]: updated type definition

File: client/odysseus/context/index.tsx [hunk 3]
```diff
@@ -9,25 +9,52 @@ export type Nudge = {
 	initialMessage: string;
 	context?: Record< string, unknown >;
 };
+
+export type MessageRole = 'user' | 'bot';
+
+export type MessageType = 'message' | 'action' | 'meta' | 'error';
```
Comment (kthai15) [inline client/odysseus/context/index.tsx:15]: Added MessageType
https://github.com/Automattic/ai-services/blob/trunk/odysseus/app/schemas.py#L8-L12

File: client/odysseus/context/index.tsx [hunk 4]
```diff
@@ -9,25 +9,52 @@ export type Nudge = {
...
+
+export type Chat = {
+	chatId?: string | null;
+	messages: Message[];
+};
+
 interface OdysseusAssistantContextInterface {
-	currentView: string;
-	setCurrentView: ( currentView: string ) => void;
+	addMessage: ( message: Message ) => void;
```
Comment (kthai15) [inline client/odysseus/context/index.tsx:30]: Simplified params to use Message type

**Review by kthai15 — APPROVED** [review]
I pushed up a few commits to sync some types between client and server, hopefully they make sense 😃

**Comment by AllTerrainDeveloper** [comment]
More than welcome those changes!

**Review by gravityrail — APPROVED** [review]
This LGTM

**Pull request merged**